sabnzbd/nzbstuff.py

```diff
@@ -864,6 +864,24 @@ def __init__(
                 self.pause()
                 self.priority = NORMAL_PRIORITY
 
+        # Check if there are any unwanted extension in plain sight in the NZB itself
+        for nzf in self.files:
+            if (
+                cfg.action_on_unwanted_extensions() >= 1
+                and get_ext(nzf.filename).replace(".", "") in cfg.unwanted_extensions()
+            ):
+                # So we found an unwanted extension
+                logging.warning(T("Unwanted Extension in file %s (%s)") % (nzf.filename, self.final_name))
+                # Pause, or Abort:
+                if cfg.action_on_unwanted_extensions() == 1:
+                    logging.debug("Unwanted extension ... pausing")
+                    self.unwanted_ext = 1
+                    self.pause()
+                if cfg.action_on_unwanted_extensions() == 2:
+                    logging.debug("Unwanted extension ... aborting")
+                    self.fail_msg = T("Unwanted extension ... aborting")
```

Comment:
I would use `T("Aborted, unwanted extension detected")`, it's a better history-status and it's already an available translated-texts from `assembler.py`.